rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/serialization/CatalogRequestBuilder.java

```diff
@@ -0,0 +1,107 @@
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+
+
+public class CatalogRequestBuilder {
+
+    private final CatalogObject catalogObject;
+
+    public CatalogRequestBuilder(CatalogObject catalogObject) {
+        this.catalogObject = catalogObject;
+    }
+
+    public String build() {
+        return new StringBuilder(URL_CATALOG).append("/buckets/")
+                                             .append(this.catalogObject.getBucketName())
+                                             .append("/resources?")
+                                             .append(paramPair(NAME_PARAM, this.catalogObject.getNodeSourceName()))
+                                             .append("&")
+                                             .append(paramPair(KIND_PARAM, this.catalogObject.getKind()))
+                                             .append("&")
+                                             .append(paramPair(COMMIT_MESSAGE_PARAM,
+                                                               this.catalogObject.getCommitMessage()))
+                                             .append("&")
+                                             .append(paramPair(OBJECT_CONTENT_TYPE_PARAM,
+                                                               this.catalogObject.getObjectContentType()))
+                                             .toString();
+    }
+
+    public void postNodeSourceRequestToCatalog(String sessionId, String fullUri, HttpServletResponse response)
+            throws NoSuchAlgorithmException, KeyStoreException, KeyManagementException, IOException {
+        HttpPost postNodeSource = buildCatalogRequest(sessionId, fullUri);
+        getHttpClientBuilder().build().execute(postNodeSource);
+    }
+
+    private String paramPair(String param, String value) {
+        return param + "=" + value;
+    }
+
+    private HttpPost buildCatalogRequest(String sessionId, String fullUri) {
+        String boundary = "---------------" + UUID.randomUUID().toString();
+        HttpPost post = new HttpPost(fullUri);
+        post.addHeader("Accept", "application/json");
+        post.addHeader("Content-Type",
+                       org.apache.http.entity.ContentType.MULTIPART_FORM_DATA.getMimeType() + ";boundary=" + boundary);
+        post.addHeader("sessionId", sessionId);
+
+        MultipartEntityBuilder builder = MultipartEntityBuilder.create();
+        builder.setBoundary(boundary);
+        builder.setMode(org.apache.http.entity.mime.HttpMultipartMode.BROWSER_COMPATIBLE);
+        builder.addPart("file", new FileBody(this.catalogObject.getNodeSourceJsonFile()));
+        post.setEntity(builder.build());
+        return post;
+    }
+
+    private HttpClientBuilder getHttpClientBuilder()
+            throws KeyStoreException, NoSuchAlgorithmException, KeyManagementException {
+        SSLContextBuilder sslContextBuilder = new SSLContextBuilder();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "SSLContextBuilder"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "SSLContextBuilder"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)